componentified-todo-list/prdaneshi/to-do-list/src/index.js

```diff
@@ -28,35 +29,37 @@ const App = () => {
   };
 
   const checkBoxClicked = (e, index) => {
-    console.log(index);
-    items[index].status = e.target.checked;
+    // console.log(index);
+    let newArray = [...items];
+    newArray[index].status = e.target.checked;
     if (e.target.checked) {
-      const checked = items.splice(index, 1);
-      items.push(checked[0]);
+      const checked = newArray.splice(index, 1);
+      newArray.push(checked[0]);
     }
-    setItem([...items]);
+    setItem([...newArray]);
   };
 
   const DeleteTask = (e, index) => {
-    items.splice(index, 1);
-    setItem([...items]);
+    let newArray = [...items];
+    newArray.splice(index, 1);
+    setItem([...newArray]);
   };
 
-  const [inputValue, setinputValue] = useState(0);
   const InputValue = (e) => {
     setinputValue(e.target.value);
   };
 
   const CreatList = () => {
     const result = Object.entries(items).map((item) => (
       <List
-        id={item[0]}
+        id={new Date()}
```

Comment:
__[HIGH]__  
This doesn't work. It generates a new `id` on each render. You should generate `id`s in handlers.